File: tests/fixtures/edgecase_repo/javascript/src/module1.js
const { Service0 } = require("./module0");

class Base1 {
  constructor(name) {
    this.name = name;
  }
}

class Service1 extends Base1 {
  handle(payload) {
    this.validate(payload);
    return new Service0("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper1(value) {
  return value;
}

module.exports = { Service1 };

File: tests/fixtures/edgecase_repo/rust/src/module1.rs
use crate::module0::Service0;

pub struct Base1 {
    pub name: String,
}

pub struct Service1 {
    pub base: Base1,
}

pub trait Handler1 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler1 for Service1 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service1 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_1(value: &str) -> &str {
    value
}
